engine/rpcserver.go

```diff
@@ -1213,3 +1213,36 @@ func (s *RPCServer) GetAuditEvent(ctx context.Context, r *gctrpc.GetAuditEventRe
 
 	return &resp, nil
 }
+
+// GetHistoricCandles returns historical candles for a given exchange
+func (s *RPCServer) GetHistoricCandles(ctx context.Context, req *gctrpc.GetHistoricCandlesRequest) (*gctrpc.GetHistoricCandlesResponse, error) {
+	if req.Exchange == "" {
+		return nil, errors.New(errExchangeNameUnset)
+	}
+
+	if req.Pair.String() == "" {
+		return nil, errors.New(errCurrencyPairUnset)
+	}
+
+	candles, err := GetExchangeByName(req.Exchange).GetHistoricCandles(currency.Pair{
```

Comment:
GetExchangeByName() can return nil which might cause this to crash if an invalid exchange name is ever passed
https://github.com/thrasher-corp/gocryptotrader/blob/4e05ad41e35b248161dfa7c16f37c8a3d53e866e/engine/exchange.go#L74-L81

While this _should_ never happen due to the name check in the cli its best to also handle it just incase

Reply:
this actually happened to me: I was trying to call the RPC service from my application and didn't notice I was passing a not existent exchange name. I was getting a weird nil, took me a while to figure out what the issue was as GCT was crashing during unmarshalling and was not able to get to the actual point of failure using the debugger. Eventually,
 I just noticed the exchange name was wrong and fixed it

Reply:
Yeah i was considering at one point changing GetExchangeByName to return an error that the exchange wasn't found as well as pointer to IBotExchange